newsrevenuehub/rev-engine
# DEV-3330: Add max pages published modal

Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

N/A

#### What's this PR do?

- Adds a userCanPublish function to the useContributionPageList hook that consults a yet-to-be-created property in the backend
- Changes the Publish button so that it shows a new modal if the user has hit their plan limit
- Refactors out the published popover shown on a published page into a separate component for clarity
- Fixes bugs noted in https://news-revenue-hub.atlassian.net/browse/DEV-3366 and https://news-revenue-hub.atlassian.net/browse/DEV-3338
- Adds a Link base component for consistency in link appearance

#### Why are we doing this? How does it help us?

Adds needed functionality around the Core plan.

#### How should this be manually tested? Please include detailed step-by-step instructions.

To test that publishing works as usual:

1. Create a new page in an organization that is under the published page limit.
2. Publish the page using the Publish button and confirm the page is available, and nothing has changed from previous behavior.

To test that the new modal appears when the publish limit is hit:

1. Create a new page in an organization that has hit the published page limit.
2. Try to publish the page. Confirm that a modal appears explaining that the user has hit the page limit.
3. Confirm that the link in the modal goes to the pricing page, and that the modal can be closed.

To test that the publish modal closes on successful publish (https://news-revenue-hub.atlassian.net/browse/DEV-3366):

1. Create a new page or find an unpublished page.
2. Publish it.
3. Confirm that when you close the "Successfully published" modal, you're back at the page editor (e.g. the original publish modal asking for the slug is gone).

To test that the tooltip re: Stripe only shows when Stripe is not connected:

1. Create a new user.
2. Create a new page.
3. Confirm the publish button is disabled and that a tooltip asking the user to connect to Stripe appears when you point at it.
4. Set up Stripe for this user.
5. Confirm that the publish button is no longer disabled and that no tooltip appears when you point at it.
6. Publish the page.
7. Confirm that the publish button still isn't disabled (it should read "Published" now) and no tooltip appears when you point at it.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

We should document what happens when you hit your plan limit.

#### Have automated unit tests been added? If not, why?

Yes.

#### How should this change be communicated to end users?

See doc comment.

#### Are there any smells or added technical debt to note?

I haven't implemented AC 7 yet (e.g. showing a better error to Hub admins if they exceed the page's organization's limit). I need DEV-3328 to be done first so I know what the error message coming back will look like.

#### Has this been documented? If so, where?

No.

#### What are the relevant tickets? Add a link to any relevant ones.

- https://news-revenue-hub.atlassian.net/browse/DEV-3330
- https://news-revenue-hub.atlassian.net/browse/DEV-3366
- https://news-revenue-hub.atlassian.net/browse/DEV-3338
- https://news-revenue-hub.atlassian.net/browse/DEV-3328

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

⚠️ This should not be merged until DEV-3328 is done and this PR is tested against it.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

No.
